bindgen/codegen/serialize.rs

```diff
@@ -0,0 +1,234 @@
+                let signature_type = ctx.resolve_type(function.signature());
+                match signature_type.kind() {
+                    TypeKind::Function(signature) => {
+                        let mut buf = String::new();
+
+                        let mut count = 0;
+
+                        let name = function.name();
+                        let args = signature
+                            .argument_types()
+                            .iter()
+                            .cloned()
+                            .map(|(opt_name, type_id)| {
+                                (
+                                    opt_name.unwrap_or_else(|| {
+                                        let name = format!("arg_{}", count);
+                                        count += 1;
+                                        name
+                                    }),
+                                    type_id,
+                                )
+                            })
+                            .collect::<Vec<_>>();
+
+                        serialize_type(signature.return_type(), ctx, &mut buf)?;
+                        write!(
+                            buf,
+                            " {}{}(",
+                            name,
+                            ctx.options()
+                                .wrap_non_extern_fns_suffix
+                                .as_deref()
+                                .unwrap_or(DEFAULT_NON_EXTERN_FNS_SUFFIX)
+                        )?;
+                        serialize_sep(
+                            ", ",
+                            args.iter(),
+                            ctx,
+                            &mut buf,
+                            |(name, type_id), ctx, buf| {
+                                serialize_type(*type_id, ctx, buf)?;
+                                write!(buf, " {}", name).map_err(Error::from)
+                            },
+                        )?;
+                        write!(buf, ")")?;
+
+                        let header = format!("{};", buf);
+
+                        write!(buf, " {{ return {}(", name)?;
+                        serialize_sep(
+                            ", ",
+                            args.iter(),
+                            ctx,
+                            &mut buf,
+                            |(name, _), _, buf| {
+                                write!(buf, "{}", name).map_err(Error::from)
+                            },
+                        )?;
+                        write!(buf, "); }}")?;
+
+                        Ok(Self { header, code: buf })
+                    }
+                    _ => unreachable!(),
+                }
+            }
+            function_kind => Err(Error::Serialize(format!(
+                "Cannot serialize function kind {:?}",
+                function_kind
+            ))),
+        }
+    }
+
+    pub(crate) fn header(&self) -> &str {
+        self.header.as_ref()
```

Comment:
Do we even need to generate the header at this point? We don't need to parse it or anything, so I think we can just generate the C/CPP file?